From aed621d5db1d521404238a92f16b879180a469a2 Mon Sep 17 00:00:00 2001
From: Michel Alexandre Salim <michel@michel-slm.name>
Date: Wed, 27 Jan 2021 20:32:07 -0800
Subject: [PATCH] Dynamically detect libdir on Linux

Some Linux distributions (e.g. Fedora, CentOS) put 64-bit libraries
in `/usr/lib64` rather than `/usr/lib`. On such systems `luarocks`
should use `lib64` rather than `lib`.

Signed-off-by: Michel Alexandre Salim <michel@michel-slm.name>
---
 src/luarocks/core/cfg.lua |   12 ++++++++++++
 1 file changed, 12 insertions(+)

Index: luarocks-3.13.0/src/luarocks/core/cfg.lua
===================================================================
--- luarocks-3.13.0.orig/src/luarocks/core/cfg.lua	2025-12-29 00:01:55.000000000 +0100
+++ luarocks-3.13.0/src/luarocks/core/cfg.lua	2026-01-06 07:56:43.952247974 +0100
@@ -410,6 +410,18 @@
       local xdg_cache_home = os.getenv("XDG_CACHE_HOME") or home.."/.cache"
       defaults.local_cache = xdg_cache_home.."/luarocks"
       defaults.web_browser = "xdg-open"
+      if platforms.linux then
+         -- inline code from fs/linux.lua since
+         -- luarocks.fs can't be required here
+         -- (circular dependencies)
+         local fd, _, code = io.open("/usr/lib64", "r")
+         if code ~= 2 then
+            defaults.lib_modules_path = "/lib64/lua/"..lua_version
+         end
+         if fd then
+            fd:close()
+         end
+      end
    end
 
    if platforms.cygwin then
